go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -489,20 +489,7 @@ func (s *VtctldServer) backupTablet(ctx context.Context, tablet *topodatapb.Tabl
 				logger.Errorf("failed to send stream response %+v: %v", resp, err)
 			}
 		case io.EOF:
-			// Do not do anything for primary tablets and when active reparenting is disabled
-			if mysqlctl.DisableActiveReparents || tablet.Type == topodatapb.TabletType_PRIMARY {
-				return nil
-			}
-
-			// Otherwise we find the correct primary tablet and set the replication source,
-			// since the primary could have changed while we executed the backup which can
-			// also affect whether we want to send semi sync acks or not.
-			tabletInfo, err := s.ts.GetTablet(ctx, tablet.Alias)
-			if err != nil {
-				return err
-			}
-
-			return reparentutil.SetReplicationSource(ctx, s.ts, s.tmc, tabletInfo.Tablet)
+			return nil
```

Comment:
a question here: if there is a reparent operation and some other replica is promoted to primary, will this replica still point to the correct new primary?

Reply:
Also, can we add a test for that as well, if it does not exist already?

Reply:
It will, as I have ported this exact logic to rpc_backup, with the addition of checking to not demote yourself if you're the primary